scalafix-reflect/src/main/scala-3/scalafix/internal/reflect/RuleCompiler.scala

```diff
@@ -0,0 +1,93 @@
+package scalafix.internal.reflect
+
+import scalafix.internal.reflect.exception.CompilationException
+
+import java.io.File
+import java.nio.file.Paths
+import scala.meta.io.AbsolutePath
+
+import dotty.tools.dotc.Compiler
+import dotty.tools.dotc.Run
+import dotty.tools.dotc.util.SourceFile
+import dotty.tools.dotc.core.Contexts.NoContext.given_Context
+import dotty.tools.dotc.reporting.StoreReporter
+
+import scala.reflect.io.VirtualDirectory
+import scala.reflect.io.AbstractFile
+import scala.reflect.internal.util.AbstractFileClassLoader
+
+import metaconfig.Configured
+import metaconfig.Input
+
+class RuleCompiler(
+    classpath: String,
+    target: AbstractFile = new VirtualDirectory("(memory)", None)
+) {
+  // we are not configuring any settings!
+  // private val settings = new Settings()
+  // settings.deprecation.value = true // enable detailed deprecation warnings
+  // settings.unchecked.value = true // enable detailed unchecked warnings
+  // settings.outputDirs.setSingleOutput(target)
+  // settings.classpath.value = classpath
+
+  private lazy val reporter: StoreReporter = new StoreReporter()
+  private val compiler: Compiler = new Compiler()
+  private val classLoader: AbstractFileClassLoader =
+    new AbstractFileClassLoader(target, this.getClass.getClassLoader)
+
+  @throws(classOf[CompilationException])
+  def compile(input: Input): Configured[ClassLoader] = {
+    val run: Run = compiler.newRun
+
+    run.compileSources(
+      List(
+        new SourceFile(
+          dotty.tools.io.AbstractFile.getFile(input.path),
```

Comment:
we are using both `dotty.tools.io.AbstractFile` and `scala.reflect.io.AbstractFile`

Reply:
personal taste, but I would import this as top level with an alias (`DottyAbstractFile`) to make the dependency clear.